src/test/java/PawnTest.java

```diff
@@ -0,0 +1,684 @@
+        Pawn test_pawn = new Pawn();
+        LocationBitboard new_board = new LocationBitboard();
+        TestHelper.remove_all_pieces(new_board);
+        int[][] arrayfrom = {
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 1, 0, 0, 0, 0}
+        };
+        new_board.blackPawn[0] = TestHelper.bitboard_representation(arrayfrom);
+        new_board.updateLocationVariables();
+        long from = TestHelper.bitboard_representation(arrayfrom);
+        long expected_moves = 0L;
+        long test_moves = test_pawn.valid_moves(from, false, new_board);
+        assertEquals(expected_moves, test_moves);
+    }
+
+    // en passant capture
+    @Test
+    public void en_passant_capture_testb() {
+        Pawn test_pawn = new Pawn();
+        LocationBitboard new_board = new LocationBitboard();
+        TestHelper.remove_all_pieces(new_board);
+        int[][] arrayfrom = {
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 0, 1, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0}
+        };
+        int[][] arrayto = {
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 1, 1, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0}
+        };
+        int[][] enemy_pawn_from = {
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 1, 0, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0}
+        };
+        int[][] enemy_pawn_to = {
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 1, 0, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0}
+        };
+        new_board.blackPawn[0] = TestHelper.bitboard_representation(arrayfrom);
+        new_board.whitePawn[0] = TestHelper.bitboard_representation(enemy_pawn_from);
+        new_board.updateLocationVariables();
+        long black_pawn_from_long = TestHelper.bitboard_representation(enemy_pawn_from);
```

Comment:
Should be named white_pawn_from_long since it represents a white piece in this case.